src/main/java/org/opensearch/plugin/insights/core/exporter/LocalIndexExporter.java

```diff
@@ -113,4 +119,47 @@ public void close() {
     private String getDateTimeFromFormat() {
         return indexPattern.print(DateTime.now(DateTimeZone.UTC));
     }
+
+    /**
+     * Set local index exporter data retention period
+     *
+     * @param deleteAfter the number of days after which Top N local indices should be deleted
+     */
+    public void setDeleteAfter(final int deleteAfter) {
+        this.deleteAfter = deleteAfter;
+    }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    public void deleteExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
+        long expirationMillisLong = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(deleteAfter);
+        for (IndexMetadata indexMetadata : indexMetadataMap.values()) {
+            String indexName = indexMetadata.getIndex().getName();
+            if (!matchesPattern(indexName, indexPattern)) {
+                continue;
+            }
+            if (indexMetadata.getCreationDate() <= expirationMillisLong) {
+                // delete this index
+                client.admin().indices().delete(new DeleteIndexRequest(indexMetadata.getIndex().getName()));
```

Comment:
`client.admin().indices().delete()` returns a `ActionFuture`, we should capture the delete results, if there are errors/exceptions, we should record the them into an operational metric, or even do some retries. 
something like
```
client.admin().indices().delete(deleteRequest, new ActionListener<>() {
                    @Override
                    public void onResponse(AcknowledgedResponse response) {
                        if (response.isAcknowledged()) {
                            // pass
                        } else {}
                    }

                    @Override
                    public void onFailure(Exception e) {
                       // log in OperationalMetric, retry for one time
                    }
                });
```

Reply:
I had the ActionListener but checkstyle would not allow me to import the AcknowledgedResponse class. I saw Query-Insights has some extra github check that is not part of OpenSearch core.